imod/msw/utilities/parse.py

```diff
@@ -0,0 +1,35 @@
+from pathlib import Path
+from typing import TypeAlias
+
+ScalarType: TypeAlias = str | float | int
+
+
+def _try_parsing_string_to_number(s: str) -> ScalarType:
+    """
+    Convert string to number:
+
+    "1" -> 1
+    "1.0" -> 1.0
+    "a" -> "a"
+    """
+    try:
```

Comment:
I wouldn't nest try-except statements.
Im suprised there is not an exisiting python method which does this.

An alternative:
```
try:
  return int(s)
except ValueError:
  pass

try:
  return float(s)
except ValueError:
  pass

return s
```